src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp

```diff
@@ -717,7 +721,11 @@ bool SlaterDetBuilder::createMSDFast(MultiSlaterDeterminantFast* multiSD, xmlNod
       {
         //std::stringstream sstr;
         //sstr << "CIcoeff" << "_" << i;
-        multiSD->myVars->insert(CItags[i], (*(multiSD->CSFcoeff))[i], true, optimize::LINEAR_P);
+        #ifdef QMC_COMPLEX
+          multiSD->myVars->insert(CItags[i], std::real((*(multiSD->CSFcoeff))[i]), true, optimize::LINEAR_P);
```

Comment:
I guess this is just a workaround to get the code building.
Could you add a note
```
// TODO: to be fixed by the complex VariableSet.
```
Could you add such TODO notes in all the places needing a fix later.

Reply:
@ye-luo I have added the note.

Reply:
Is this the only place? Or you have others in mind. Please do all places.

Reply:
Should be done by now.